feat: allow disabling selected item events

korczas opened the pull request:
<!--
Thanks for your interest in the project. I appreciate bugs filed and PRs submitted!
Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Please fill out this template with all the relevant information so we can
understand what's going on and fix the issue.

I'll probably ask you to submit the fix (after giving some direction). If you've
never done that before, that's great! Check this free short video tutorial to
learn how: http://kcd.im/pull-request
-->

- `downshift` version: latest
- `node` version: 12.16.3
- `npm` version: 6.14.4
- `yarn` version: 1.22.10

**Relevant code or config**

https://codesandbox.io/s/priceless-tree-5e5bj?file=/src/hooks/useMultipleSelection/combobox.js

```javascript
import React, {useState} from 'react'
import {render} from 'react-dom'
import {useCombobox, useMultipleSelection} from 'downshift'
import {
  items,
  menuMultipleStyles,
  comboboxStyles,
  comboboxWrapperStyles,
  selectedItemStyles,
  selectedItemIconStyles,
} from '../../shared'

const DISABLED = true

function DropdownMultipleCombobox() {
  const [inputValue, setInputValue] = useState('')
  const {
    getSelectedItemProps,
    getDropdownProps,
    addSelectedItem,
    removeSelectedItem,
    selectedItems,
  } = useMultipleSelection({initialSelectedItems: [items[0], items[1]]})
  const getFilteredItems = (items) =>
    items.filter(
      (item) =>
        selectedItems.indexOf(item) < 0 &&
        item.toLowerCase().startsWith(inputValue.toLowerCase()),
    )
  const {
    isOpen,
    getToggleButtonProps,
    getLabelProps,
    getMenuProps,
    getInputProps,
    getComboboxProps,
    highlightedIndex,
    getItemProps,
    selectItem,
  } = useCombobox({
    inputValue,
    items: getFilteredItems(items),
    onStateChange: ({inputValue, type, selectedItem}) => {
      switch (type) {
        case useCombobox.stateChangeTypes.InputChange:
          setInputValue(inputValue)
          break
        case useCombobox.stateChangeTypes.InputKeyDownEnter:
        case useCombobox.stateChangeTypes.ItemClick:
        case useCombobox.stateChangeTypes.InputBlur:
          if (selectedItem) {
            setInputValue('')
            addSelectedItem(selectedItem)
            selectItem(null)
          }

          break
        default:
          break
      }
    },
  })
  return (
    <div>
      <label {...getLabelProps()}>Choose some elements:</label>
      <div style={comboboxWrapperStyles}>
        {selectedItems.map((selectedItem, index) => (
          <span
            style={selectedItemStyles}
            key={`selected-item-${index}`}
            {...getSelectedItemProps({selectedItem, index, disabled: DISABLED})}
          >
            {selectedItem}
            <span
              style={selectedItemIconStyles}
              onClick={() =>
                DISABLED ? undefined : removeSelectedItem(selectedItem)
              }
            >
              &#10005;
            </span>
          </span>
        ))}
        <div style={comboboxStyles} {...getComboboxProps()}>
          <input
            {...getInputProps(
              getDropdownProps({preventKeyAction: isOpen, disabled: DISABLED}),
            )}
          />
          <button
            {...getToggleButtonProps({disabled: DISABLED})}
            aria-label={'toggle menu'}
          >
            &#8595;
          </button>
        </div>
      </div>
      <ul {...getMenuProps()} style={menuMultipleStyles}>
        {isOpen &&
          getFilteredItems(items).map((item, index) => (
            <li
              style={
                highlightedIndex === index ? {backgroundColor: '#bde4ff'} : {}
              }
              key={`${item}${index}`}
              {...getItemProps({item, index})}
            >
              {item}
            </li>
          ))}
      </ul>
    </div>
  )
}

render(<DropdownMultipleCombobox />, document.getElementById('root'))

```

**What you did**:
I have disabled edition whenever I could, but there is still an option to delete the items. 

**What happened**:
You can click on item and then <Esc> or <Backspace> to delete the items.

**Suggested solution**:
see PR

**FINAL NOTES**
I wasn't able to correctly commit due to multiple node errors (when I used yarn to install packages then `kcd-scripts lint` failed with `Environment key "jest/globals" is unknown`, and if I used npm then `npm run test:cypress --silent exited with code SIGTERM` 😔). At the end I used `--no-verify` flag to commit.

silviuaavram commented:
Hi! We don't support `disabled` on `useMultipleSelection` but I think it's a good idea to have it! Can you create a Feature Request?

korczas commented:
@silviuaavram Hi. I am not sure if I understood correctly what u meant by creating a Feature Request, by I have updated PR description with the template, added TS definition and updated docs. It seems that flow doesn't need any update, but I might be wrong.

korczas commented:
@silviuaavram Hi, feature is ready to ship.

Lelith commented:
Hey i wanted to ask what the status of this pull request is, as i would be also interested in the possibility to disable some of my elements from a list for selection.

korczas commented:
@silviuaavram

silviuaavram commented:
Well yes, all good so far, thank you! But if the item is disabled, it should be skipped during navigation. This PR needs more work and tests.
